schedule/lvm_thin_provisioning.yml

```diff
@@ -0,0 +1,77 @@
+name:           lvm_thin_provisioning
+description:    >
+    Complete OS deployment with unencrypted LVM drive management.
+    Test creates 2 LVM and BIOS boot partitions.
+    Thin pool and thin lv resides on the second LVM partition,
+    where /home (XFS) is being mounted. Maintainer: mloviska, poo#39023
+conditions:
+    bootloader:
+        ARCH:
+            x86_64:
+                - installation/isosize
+                - installation/bootloader
+            s390x:
+                - installation/bootloader_zkvm
+            aarch64:
```

Comment:
could be combined in yaml with `x86_64`? something like `[x86_64, aarch64]`. Not sure about exact syntax.

Reply:
Those are keys in the hash, so that would complicate parsing. So I would like than to hear alternative yaml structure. But yes, it shows those issues and this approach cannot work for e.g. default test suite where we have even more differences on the same architecture. I believe we should act on these in a different way rather having those conditions in the schedule to keep it stupid simple which is the goal.

Reply:
And actually bootloader for arm is different. It's `bootloader_uefi`. But yeah, it's easy to introduce some default schedule when non of the keys matched, so common part can be extracted and only exceptions are mentioned.

Reply:
Problem I see though is that in the past people have introduced differences depending on the ARCH where actually the difference lies elsewhere and I would suggest to use the according conditions. `aarch64` uses `bootloader_uefi` not because it's `aarch64` but because it's `UEFI`. On `x86_64` we have both, `UEFI=1` using `bootloader_uefi` and `UEFI=0` using `bootloader`. How can we encode that?

Reply:
There are anchors and references in yaml for DRY, (and they would still be keys) but it needs to be checked visually if it really helps, with a few items does not worth at all, and to name the anchor is also tricky, for example, it would be something like:
```
x86_64: &bootloader_common
          - installation/isosize
          - installation/bootloader
    aarch64: *bootloader_common
```

Reply:
YAML is not a markup language, there are a lot of features which people have introduced on top of it, including imports of other yaml files. And DRY is not applicable here, as those are different strings. So, yes, we can introduce anchors if we want (I guess we will have to, as well as yml file imports). In general we could introduce some keyword `default`, so with condition we will handle exceptional cases instead, which is better fit for the exact problem here, as I doubt that anyone wants to mention all the cases with reference to the same block.